apptest/tests/maxingestionrate_test.go

```diff
@@ -0,0 +1,35 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/apptest"
+)
+
+func TestSingleMaxIngestionRateIncrementsMetric(t *testing.T) {
+	tc := apptest.NewTestCase(t)
+	defer tc.Stop()
+	sut := tc.MustStartVmsingle("vmsingle", []string{"-maxIngestionRate=1"})
+	sut.PrometheusAPIV1ImportPrometheus(t, docData, apptest.QueryOpts{})
+	maxIngestionRateMetric := sut.GetMetric(t, "vm_max_ingestion_rate_limit_reached_total")
+	if maxIngestionRateMetric <= 0 {
+		t.Errorf("Max Ingestion Rate metric not set unexpectedly")
+	} else {
+		t.Logf("MAX INGEST RATE HIT %d times", int(maxIngestionRateMetric))
+	}
+	sut.ForceFlush(t)
```

Comment:
ForceFlush() is normally put right after insertion. Also, since you don't select the data afterwards, you don't need to force flush.